src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java

```diff
@@ -158,14 +280,42 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
         return buildStepResult;
     }
 
-    // Public but restricted so we can add tests without completely changing the tests package
-    @Restricted(value=org.kohsuke.accmod.restrictions.NoExternalUse.class)
-    public String getProjectListAsString(List<Job> projectList){
+    private static String getOrCreateHyperlinkNote(Map<Object, String> hyperlinkNotes, Run run)
+    {
+      String note = hyperlinkNotes.get(run);
+      if (note == null) {
+        /** prevent calling encodeTo to often, it uses gzip deflate which takes long to init */
+        note = HyperlinkNote.encodeTo('/' + run.getUrl(), run.getFullDisplayName());
+        hyperlinkNotes.put(run, note);
+      }
+      return note;
+    }
+
+    private static String getOrCreateHyperlinkNote(Map<Object, String> hyperlinkNotes, Job job)
+    {
+      String note = hyperlinkNotes.get(job);
+      if (note == null) {
+        /** prevent calling encodeTo to often, it uses gzip deflate which takes long to init */
+        note = HyperlinkNote.encodeTo('/' + job.getUrl(), job.getFullDisplayName());
+        hyperlinkNotes.put(job, note);
+      }
+      return note;
+    }
+
+    
+    // Public but restricted so we can add tests without completely changing the
+    // tests package
+    @Restricted(value = org.kohsuke.accmod.restrictions.NoExternalUse.class)
+    public String getProjectListAsString(List<Job> projectList) {
+      return getProjectListAsString(new HashMap<Object, String>(), projectList);
+    }
+
+    public String getProjectListAsString(Map<Object, String> hyperlinkNotes, List<Job> projectList) {
```

Comment:
@kwhetstone 
Do you have a hint which existing testcase that is? I know how to unit test in general however I never did with jenkins. Any suggestions where to start?